parser/ast/functions.go

```diff
@@ -380,21 +381,9 @@ type FuncCallExpr struct {
 
 // Restore implements Node interface.
 func (n *FuncCallExpr) Restore(ctx *format.RestoreCtx) error {
-	var specialLiteral string
-	switch n.FnName.L {
-	case DateLiteral:
-		specialLiteral = "DATE "
-	case TimeLiteral:
-		specialLiteral = "TIME "
-	case TimestampLiteral:
-		specialLiteral = "TIMESTAMP "
-	}
-	if specialLiteral != "" {
-		ctx.WritePlain(specialLiteral)
-		if err := n.Args[0].Restore(ctx); err != nil {
-			return errors.Annotatef(err, "An error occurred while restore FuncCastExpr.Expr")
-		}
-		return nil
+	done, err := n.customRestore(ctx)
```

Comment:
Could this `customRestore` be inlined? (Or why is it seperated?) We could `return nil` at the end of the `n.FnName.L == JSONMemberOf` condition :thinking: .

Reply:
just think it will be more clear.

Reply:
OK